VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionCommentsAdapter.java

```diff
@@ -107,7 +144,34 @@ public void run() {
 
     @Override
     public int getItemCount() {
```

Comment:
Yeah, the (not so well named) `progressVisible` refers to the loading indicator, which is added as an extra item in the `RecyclerView` adapter (this class). This method returns the total number of items to be displayed by the `RecyclerView`, which is the sum of the comments count, the parent response header, and the loading indicator. That does introduce some complexity in the adapter when calculating offsets and count.

The `InfiniteScrollUtils` utility class adds a footer view for displaying the loading indicator in the case of `ListView` to take the logic out of the adapter, but unfortunately `RecyclerView` doesn't define methods for adding header or footer items. I guess we can introduce a wrapper class for the adapter that handles this behaviour, and saves us from having to define it in each individual adapter implementation. Since this affects multiple modules beyond the scope of the changes here, I'll open another pull request to introduce this change, and let you evaluate it.

As for unit testing, we should also implement it for the whole `InfiniteScrollUtils` implementation at some point (along with the other classes), but I guess this is a larger task that can be implemented later.